<launch>


    <arg name="front_laser_port_name"  default="/dev/front_laser"/>
    <arg name="front_laser_frame_id"   default="front_laser_frame"/>
    <arg name="front_laser_topic"   default="front_scan"/>
    <arg name="front_node_name"   default="front_ydlidar_node"/>

    <arg name="front_angle_min"  default="-180"/>
    <arg name="front_angle_max"   default="180"/>

    <arg name="rear_laser_port_name"  default="/dev/rear_laser"/>
    <arg name="rear_laser_frame_id"   default="rear_laser_frame"/>
    <arg name="rear_laser_topic"   default="rear_scan"/>
    <arg name="rear_node_name"   default="rear_ydlidar_node"/>

    <arg name="rear_angle_min"  default="-180"/>
    <arg name="rear_angle_max"   default="180"/>

    <!--bringup the imu -->
    <include file="$(find scholar_r500_imu_bringup)/launch/scholar_r500_imu_bringup.launch"/>

    <!--bringup the scholar r500 robot-->
    <include file="$(find scholar_r500_base_driver)/launch/scholar_r500_base_bringup.launch"/>

    <include file="$(find scholar_r500_description)/launch/display.launch"/>

    <!--bringup the lidar-->
    <include file="$(find scholar_r500_navigation)/launch/g4_laser_bringup.launch">

        <arg name="port_name" value="$(arg front_laser_port_name)"/>
        <arg name="frame_id" value="$(arg front_laser_frame_id)"/>
        <arg name="pub_topic" value="$(arg front_laser_topic)"/>
        <arg name="node_name"   value="$(arg front_node_name)"/>
        <arg name="angle_min"   value="$(arg front_angle_min)"/>

        <arg name="angle_max"   value="$(arg front_angle_max)"/>

    </include>

    <include file="$(find scholar_r500_navigation)/launch/g4_laser_bringup.launch">

        <arg name="port_name" value="$(arg rear_laser_port_name)"/>
        <arg name="frame_id" value="$(arg rear_laser_frame_id)"/>
        <arg name="pub_topic" value="$(arg rear_laser_topic)"/>
        <arg name="node_name"   value="$(arg rear_node_name)"/>

        <arg name="angle_min"  value="$(arg rear_angle_min)"/>
        <arg name="angle_max"   value="$(arg rear_angle_max)"/>

    </include>


</launch>
